devices/danfoss.js

```diff
@@ -14,7 +14,7 @@ module.exports = [
         model: '014G2461',
         vendor: 'Danfoss',
         description: 'Ally thermostat',
-        fromZigbee: [fz.battery, fz.legacy.thermostat_att_report, fz.hvac_user_interface, fz.danfoss_thermostat],
+        fromZigbee: [fz.battery, fz.thermostat, fz.hvac_user_interface, fz.danfoss_thermostat],
```

Comment:
I believe this will introduce a breaking change, note that `fz. thermostat` will be used whenever you set `legacy: false` or for all devices

```yaml
device_options:
  legacy: false
```

Reply:
@Koenkk Hopefully I understand how this operates that my judgement was correct.

As you know from before, I own the Hive TRV with is the same device as Danfoss Ally, and I have improved both of them a while ago based on the Danfoss Ally specs.

Since then, I have used my TRVs with legacy: false, and all commands and attributes behave correctly.

On that premise, I have removed the legacy converter and replaced with the new fz.thermostat. I have paired the TRV with these new settings and everything works just like before.

Would you be kind to explain further where do you foresee the break to be? As in, in what context or setup?

Here is logs from a few minutes ago of me re-pairing the device and pressing all the buttons:
```
Zigbee2MQTT:info  2021-08-07 12:17:42: Zigbee: allowing new devices to join.
Zigbee2MQTT:info  2021-08-07 12:17:42: MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"cg1jd-1"}'
Zigbee2MQTT:info  2021-08-07 12:17:51: Device '0x842e14fffe2910dc' joined
Zigbee2MQTT:info  2021-08-07 12:17:51: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x842e14fffe2910dc","ieee_address":"0x842e14fffe2910dc"},"type":"device_joined"}'
Zigbee2MQTT:info  2021-08-07 12:17:51: Starting interview of '0x842e14fffe2910dc'
Zigbee2MQTT:info  2021-08-07 12:17:51: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x842e14fffe2910dc","ieee_address":"0x842e14fffe2910dc","status":"started"},"type":"device_interview"}'
Zigbee2MQTT:debug 2021-08-07 12:18:23: Device '0x842e14fffe2910dc' announced itself
Zigbee2MQTT:info  2021-08-07 12:18:23: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x842e14fffe2910dc","ieee_address":"0x842e14fffe2910dc"},"type":"device_announce"}'
Zigbee2MQTT:debug 2021-08-07 12:18:29: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"modelId":"TRV001"}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:31: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"manufacturerName":"Danfoss"}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:32: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"powerSource":3}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:34: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"zclVersion":3}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:35: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"appVersion":33}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:37: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"stackVersion":8}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:38: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"hwVersion":69}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:40: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"dateCode":"20200402"}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:18:41: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genBasic', data '{"swBuildId":"02.33.0008 02.33"}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:18:41: Successfully interviewed '0x842e14fffe2910dc', device has successfully been paired
Zigbee2MQTT:info  2021-08-07 12:18:41: Device '0x842e14fffe2910dc' is supported, identified as: Hive Radiator valve based on Danfos Ally (UK7004240)
Zigbee2MQTT:info  2021-08-07 12:18:41: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Radiator valve based on Danfos Ally","exposes":[{"access":1,"description":"Remaining battery in %","name":"battery","property":"battery","type":"numeric","unit":"%","value_max":100,"value_min":0},{"access":7,"description":"Enables/disables physical input on the device","name":"keypad_lockout","property":"keypad_lockout","type":"binary","value_off":"unlock","value_on":"lock1"},{"access":5,"description":"Is the unit in mounting mode. This is set to `false` for mounted (already on the radiator) or `true` for not mounted (after factory reset)","name":"mounted_mode_active","property":"mounted_mode_active","type":"binary","value_off":false,"value_on":true},{"access":7,"description":"Set the unit mounting mode. `false` Go to Mounted Mode or `true` Go to Mounting Mode","name":"mounted_mode_control","property":"mounted_mode_control","type":"binary","value_off":false,"value_on":true},{"access":7,"description":"Thermostat Orientation. This is important for the PID in how it assesses temperature. `false` Horizontal or `true` Vertical","name":"thermostat_vertical_orientation","property":"thermostat_vertical_orientation","type":"binary","value_off":false,"value_on":true},{"access":7,"description":"Viewing/Display Direction. `0` Horizontal or `1` Vertical","name":"viewing_direction","property":"viewing_direction","type":"numeric","value_max":1,"value_min":0},{"access":7,"description":"Not clear how this affects operation. `false` No Heat Available or `true` Heat Available","name":"heat_available","property":"heat_available","type":"binary","value_off":false,"value_on":true},{"access":5,"description":"Whether or not the unit needs warm water. `false` No Heat Request or `true` Heat Request","name":"heat_required","property":"heat_required","type":"binary","value_off":false,"value_on":true},{"features":[{"access":7,"description":"Temperature setpoint","name":"occupied_heating_setpoint","property":"occupied_heating_setpoint","type":"numeric","unit":"°C","value_max":32,"value_min":5,"value_step":0.5},{"access":5,"description":"Current temperature measured on the device","name":"local_temperature","property":"local_temperature","type":"numeric","unit":"°C"},{"access":1,"description":"Position of the valve (= demanded heat) where 0% is fully closed and 100% is fully open","name":"pi_heating_demand","property":"pi_heating_demand","type":"numeric","unit":"%","value_max":100,"value_min":0},{"access":7,"description":"Mode of this device","name":"system_mode","property":"system_mode","type":"enum","values":["heat"]}],"type":"climate"},{"access":5,"description":"0=Quarantine, 1=Windows are closed, 2=Hold - Windows are maybe about to open, 3=Open window detected, 4=In window open state from external but detected closed locally","name":"window_open_internal","property":"window_open_internal","type":"numeric","value_max":4,"value_min":0},{"access":7,"description":"Set if the window is open or close. This setting will trigger a change in the internal window and heating demand. `false` (windows are closed) or `true` (windows are open)","name":"window_open_external","property":"window_open_external","type":"binary","value_off":false,"value_on":true},{"access":7,"description":"Exercise day of week: 0=Sun...6=Sat, 7=undefined","name":"day_of_week","property":"day_of_week","type":"enum","values":["sunday","monday","tuesday","wednesday","thursday","friday","saturday","away_or_vacation"]},{"access":7,"description":"Exercise trigger time. Minutes since midnight (65535=undefined). Range 0 to 1439","name":"trigger_time","property":"trigger_time","type":"numeric","value_max":65535,"value_min":0},{"access":7,"description":"Scale factor of setpoint filter timeconstant (\"aggressiveness\" of control algorithm) 1= Quick ...  5=Moderate ... 10=Slow","name":"algorithm_scale_factor","property":"algorithm_scale_factor","type":"numeric","value_max":10,"value_min":1},{"access":5,"description":"Load estimate on this radiator","name":"load_estimate","property":"load_estimate","type":"numeric"},{"access":1,"description":"Link quality (signal strength)","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"UK7004240","supports_ota":false,"vendor":"Hive"},"friendly_name":"0x842e14fffe2910dc","ieee_address":"0x842e14fffe2910dc","status":"successful","supported":true},"type":"device_interview"}'
Zigbee2MQTT:info  2021-08-07 12:18:41: Configuring '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:18:46: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":120}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:18:46: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"battery":60,"linkquality":178}'
Zigbee2MQTT:debug 2021-08-07 12:18:53: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"keypad_lockout":""}'
Zigbee2MQTT:debug 2021-08-07 12:18:53: Publishing get 'get' 'keypad_lockout' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:18:55: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacUserInterfaceCfg', data '{"keypadLockout":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:18:55: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"battery":60,"keypad_lockout":"unlock","linkquality":178}'
Zigbee2MQTT:debug 2021-08-07 12:18:57: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"mounted_mode_active":""}'
Zigbee2MQTT:debug 2021-08-07 12:18:57: Publishing get 'get' 'mounted_mode_active' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:18:58: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"danfossAlgorithmScaleFactor":1,"danfossDayOfWeek":4,"danfossHeatAvailable":1,"danfossMountedModeActive":1,"danfossMountedModeControl":1,"danfossTriggerTime":660,"danfossWindowOpenExternal":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:18:58: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"keypad_lockout":"unlock","linkquality":178,"mounted_mode_active":true,"mounted_mode_control":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:debug 2021-08-07 12:18:59: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"danfossMountedModeActive":1}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:18:59: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"keypad_lockout":"unlock","linkquality":178,"mounted_mode_active":true,"mounted_mode_control":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:debug 2021-08-07 12:19:00: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"thermostat_vertical_orientation":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:00: Publishing get 'get' 'thermostat_vertical_orientation' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:01: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacUserInterfaceCfg', data '{"keypadLockout":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:01: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"keypad_lockout":"unlock","linkquality":181,"mounted_mode_active":true,"mounted_mode_control":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:info  2021-08-07 12:19:01: Successfully configured '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:02: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"danfossThermostatOrientation":1}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:02: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"keypad_lockout":"unlock","linkquality":178,"mounted_mode_active":true,"mounted_mode_control":true,"thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:debug 2021-08-07 12:19:04: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"viewing_direction":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:04: Publishing get 'get' 'viewing_direction' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:06: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacUserInterfaceCfg', data '{"danfossViewingDirection":1}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:19:10: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"heat_required":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:10: Publishing get 'get' 'heat_required' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:12: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"occupied_heating_setpoint":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:12: Publishing get 'get' 'occupied_heating_setpoint' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:15: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"danfossHeatRequired":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:15: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":170,"mounted_mode_active":true,"mounted_mode_control":true,"thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:debug 2021-08-07 12:19:16: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"occupiedHeatingSetpoint":2000}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:16: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":170,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:debug 2021-08-07 12:19:18: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"local_temperature":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:18: Publishing get 'get' 'local_temperature' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:22: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"system_mode":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:22: Publishing get 'get' 'system_mode' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:24: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"window_open_internal":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:24: Publishing get 'get' 'window_open_internal' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:29: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"localTemp":2183}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:29: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":178,"local_temperature":21.83,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:debug 2021-08-07 12:19:30: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"systemMode":4}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:30: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":178,"local_temperature":21.83,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false}'
Zigbee2MQTT:debug 2021-08-07 12:19:30: Received Zigbee message from '0x842e14fffe2910dc', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":545,"imageType":66,"manufacturerCode":4678}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:19:31: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"danfossWindowOpenInternal":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:31: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":181,"local_temperature":21.83,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:19:34: Received Zigbee message from '0x842e14fffe2910dc', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":545,"imageType":66,"manufacturerCode":4678}' from endpoint 1 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:19:34: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"localTemp":2254}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:34: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":181,"local_temperature":22.54,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:19:34: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"danfossLoadEstimate":-24}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:34: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":181,"load_estimate":-24,"local_temperature":22.54,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:19:35: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"localTemp":2254}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:35: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":178,"load_estimate":-24,"local_temperature":22.54,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:19:35: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"danfossLoadEstimate":-24}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:35: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":181,"load_estimate":-24,"local_temperature":22.54,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:19:41: Received MQTT message on 'zigbee2mqtt/0x842e14fffe2910dc/get' with data '{"algorithm_scale_factor":""}'
Zigbee2MQTT:debug 2021-08-07 12:19:41: Publishing get 'get' 'algorithm_scale_factor' to '0x842e14fffe2910dc'
Zigbee2MQTT:debug 2021-08-07 12:19:54: Received Zigbee message from '0x842e14fffe2910dc', type 'readResponse', cluster 'hvacThermostat', data '{"danfossAlgorithmScaleFactor":1}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:19:54: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":178,"load_estimate":-24,"local_temperature":22.54,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:20:31: Received Zigbee message from '0x001e5e09022e1e9e', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":17127168,"imageType":521,"manufacturerCode":4153}' from endpoint 9 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:20:33: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"localTemp":2263}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:20:33: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":173,"load_estimate":-24,"local_temperature":22.63,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:20:39: Received Zigbee message from '0x001e5e09021bdaf1', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":269043264,"imageType":514,"manufacturerCode":4153}' from endpoint 5 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:20:54: Saving state to file /home/rob/Projects/zigbee2mqtt/data/state.json
Zigbee2MQTT:debug 2021-08-07 12:21:50: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"pIHeatingDemand":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:21:50: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":176,"load_estimate":-24,"local_temperature":22.63,"mounted_mode_active":true,"mounted_mode_control":true,"occupied_heating_setpoint":20,"pi_heating_demand":0,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:23:12: Received Zigbee message from '0x001e5e090220a361', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":51577862,"imageType":260,"manufacturerCode":4153}' from endpoint 9 with groupID 0
Zigbee2MQTT:debug 2021-08-07 12:23:30: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"danfossMountedModeActive":0}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:23:30: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":170,"load_estimate":-24,"local_temperature":22.63,"mounted_mode_active":false,"mounted_mode_control":true,"occupied_heating_setpoint":20,"pi_heating_demand":0,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:23:35: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"occupiedHeatingSetpoint":2200}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:23:35: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":165,"load_estimate":-24,"local_temperature":22.63,"mounted_mode_active":false,"mounted_mode_control":true,"occupied_heating_setpoint":22,"pi_heating_demand":0,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
Zigbee2MQTT:debug 2021-08-07 12:23:36: Received Zigbee message from '0x842e14fffe2910dc', type 'attributeReport', cluster 'hvacThermostat', data '{"pIHeatingDemand":31}' from endpoint 1 with groupID 0
Zigbee2MQTT:info  2021-08-07 12:23:36: MQTT publish: topic 'zigbee2mqtt/0x842e14fffe2910dc', payload '{"algorithm_scale_factor":1,"battery":60,"day_of_week":"thursday","heat_available":true,"heat_required":false,"keypad_lockout":"unlock","linkquality":168,"load_estimate":-24,"local_temperature":22.63,"mounted_mode_active":false,"mounted_mode_control":true,"occupied_heating_setpoint":22,"pi_heating_demand":31,"system_mode":"heat","thermostat_vertical_orientation":true,"trigger_time":660,"window_open_external":false,"window_open_internal":"quarantine"}'
```

Reply:
@Koenkk also, when you get a chance to look at this again, can you also please have a look at this issue and give us some feedback? https://github.com/Koenkk/zigbee2mqtt/issues/7903

It would be great if that issue can be solved for the same release as this PR. Thanks a lot

Reply:
There are some breaking difference between the legacy converter, e.g. `occupancy`:
- legacy: https://github.com/Koenkk/zigbee-herdsman-converters/blob/fd64d0396bc65fb7d0db7779d11a5b5529e222f7/lib/legacy.js#L1561
- non-legacy: https://github.com/Koenkk/zigbee-herdsman-converters/blob/fd64d0396bc65fb7d0db7779d11a5b5529e222f7/converters/fromZigbee.js#L46

Reply:
@Koenkk The device does not support occupancy. All the attributes it actually communicates about are the same in both legacy and modern. I have always used the device with legacy:false and it was fully functional. Heck, even when I added all the attributes and exposes to it I did all that and the testing with legacy: false.

I do not mind swapping back, your words is final, but the device is fully operational on the modern converters.

Reply:
> All the attributes it actually communicates about are the same in both legacy and modern.

Then this is fine indeed, thanks for the explanation!